confgenerator/testdata/invalid/linux/metrics-receiver_redis_cert_missing_key/golden_error

```diff
@@ -0,0 +1,9 @@
+[17:18] Key: 'MetricsReceiverRedis.key_file' Error:Field validation for 'key_file' failed on the 'required_with' tag
```

Comment:
Can we add a rule to https://github.com/GoogleCloudPlatform/ops-agent/blob/master/confgenerator/config.go#L85 for `required_with` validation too so that the error message is more descriptive and customized to Ops Agent config?

Something like: 'key_file' is a required field when 'cert_file' is present.

Reply:
Will be done in the follow up TLS PR.